storage/engine/rocksdb.go

```diff
@@ -563,6 +566,9 @@ func (r *rocksDBIterator) Close() {
 }
 
 func (r *rocksDBIterator) Seek(key MVCCKey) {
+	if r.batch != nil && r.batch.batch == nil {
```

Comment:
Perhaps this should be more general. Using an iterator after a snapshot has closed is also invalid. You could add an `Engine.valid()` method that is implemented by `{RocksDB,rocksDBSnapshot,rocksDBBatch}` and then verify in each of the `rocksDBIterator` calls that the associated engine is still valid. Internally `rocksDBBatch.valid` would `return r.batch != nil`.